lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntime.cpp

```diff
@@ -447,11 +447,14 @@ static bool HasReflectionInfo(ObjectFile *obj_file) {
   return hasReflectionSection;
 }
 
-SwiftLanguageRuntimeImpl::ReflectionContextInterface *
+SwiftLanguageRuntimeImpl::ThreadSafeReflectionContext 
 SwiftLanguageRuntimeImpl::GetReflectionContext() {
+  {
+    std::lock_guard<std::recursive_mutex> lock(m_add_module_mutex);
   if (!m_initialized_reflection_ctx)
-    SetupReflection();
-  return m_reflection_ctx.get();
+        SetupReflection();
```

Comment:
Why can't we sink this into `SetupReflection`? That would ensure the mutex is locked (which address the comment there) and makes the function fully self contained.

Reply:
Yeah I agree. Updated it to do that.